test/Mono.Linker.Tests/TestCasesRunner/AssemblyChecker.cs

```diff
@@ -245,15 +246,16 @@ void VerifyOverrides (MethodDefinition original, MethodDefinition linked)
 				if (overriddenMethod.Resolve () is not MethodDefinition overriddenDefinition) {
 					Assert.Fail ($"Method {linked.GetDisplayName ()} overrides method {overriddenMethod} which does not exist");
 				} else if (overriddenDefinition.DeclaringType.IsInterface) {
-					Assert.True (linked.DeclaringType.Interfaces.Select (i => i.InterfaceType).Contains (overriddenMethod.DeclaringType),
+					Assert.True (linked.DeclaringType.Interfaces.Select (i => i.InterfaceType.FullName).Contains (overriddenMethod.DeclaringType.FullName),
 						$"Method {linked} overrides method {overriddenMethod}, but {linked.DeclaringType} does not implement interface {overriddenMethod.DeclaringType}");
 				} else {
 					TypeReference baseType = linked.DeclaringType;
 					TypeReference overriddenType = overriddenMethod.DeclaringType;
 					while (baseType is not null) {
-						if (baseType.Equals (overriddenType))
+						if (baseType.FullName == overriddenType.FullName)
```

Comment:
When could this be different?

Reply:
I honestly don't remember what issue caused me to need to change this, but I think it was related to generics.